File: specification/sphere/resource-manager/Microsoft.AzureSphere/preview/2022-09-01-preview/examples-java/PostCountDevicesCatalog.java
/** Samples for Catalogs CountDevices. */
public final class Main {
    /*
     * x-ms-original-file: specification/sphere/resource-manager/Microsoft.AzureSphere/preview/2022-09-01-preview/examples/PostCountDevicesCatalog.json
     */
    /**
     * Sample code: Catalogs_CountDevices.
     *
     * @param manager Entry point to AzureSphereManager.
     */
    public static void catalogsCountDevices(com.azure.resourcemanager.sphere.AzureSphereManager manager) {
        manager.catalogs().countDevicesWithResponse("MyResourceGroup1", "MyCatalog1", com.azure.core.util.Context.NONE);
    }
}

File: specification/sphere/resource-manager/Microsoft.AzureSphere/preview/2022-09-01-preview/examples-js/PostCountDevicesCatalog.js
const { AzureSphereManagementClient } = require("@azure/arm-sphere");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Counts devices in catalog.
 *
 * @summary Counts devices in catalog.
 * x-ms-original-file: specification/sphere/resource-manager/Microsoft.AzureSphere/preview/2022-09-01-preview/examples/PostCountDevicesCatalog.json
 */
async function catalogsCountDevices() {
  const subscriptionId =
    process.env["SPHERE_SUBSCRIPTION_ID"] || "00000000-0000-0000-0000-000000000000";
  const resourceGroupName = process.env["SPHERE_RESOURCE_GROUP"] || "MyResourceGroup1";
  const catalogName = "MyCatalog1";
  const credential = new DefaultAzureCredential();
  const client = new AzureSphereManagementClient(credential, subscriptionId);
  const result = await client.catalogs.countDevices(resourceGroupName, catalogName);
  console.log(result);
}
